docs/Diagnostics.md

```diff
@@ -0,0 +1,101 @@
+### Background ###
+
+Swift's diagnostic style is inherited from [Clang][], which itself was loosely based on [GCC][]'s. These diagnostics aim to fit a terse but clear description of an issue into a single line, ideally suitable for display both on the command line and in IDEs. Additional information is provided via "notes", which are always attached to a primary diagnostic.
+
+The Clang site has [an older comparison][comparison] between Clang's diagnostics and GCC's that shows where the attention to detail has been in that compiler.
+
+This document describes several *current* guidelines for diagnostics in the Swift compiler and accompanying tools. It does not discuss potential future directions for diagnostics, such as following the examples of the [Elm][] or [Rust][] compilers.
+
+  [Clang]: http://clang.llvm.org
+  [GCC]: https://gcc.gnu.org
+  [comparison]: http://clang.llvm.org/diagnostics.html
+  [Elm]: http://elm-lang.org/blog/compiler-errors-for-humans
+  [Rust]: https://blog.rust-lang.org/2016/08/10/Shape-of-errors-to-come.html
+
+
+### Errors vs. Warnings ###
+
+Swift diagnostics are classified into errors, warnings, and notes. Notes are always considered "attached" to the immediately preceding error or warning, so the primary distinction is between errors and warnings. In the Swift compiler, a particular diagnostic should be a warning if the intent of the code is clear *and* it won't immediately result in a crash. This allows users to continue to build during refactoring without having to clean up all issues first.
+
+For a warning, consider whether there might be a legitimate need to write the code in question, even if it's rare. In this case, there should be a way to silence the warning, and a note with a fix-it suggesting this. For example, the "unused function result" warning suggests assigning to `_` to make it clear that this is intentional.
+
+Unlike Clang and many other compilers, Swift compiler warnings cannot be disabled as part of the invocation or through a standard construct in the source. This was a deliberate choice in order to keep the language from fracturing into dialects, but has been controversial throughout Swift's history.
+
+Clang also has a kind of diagnostic called a "remark", which represents information about the build that does not indicate an issue. Swift does not currently have remarks, but there's no reason why it couldn't.
+
+
+### Grammar and Phrasing ###
+
+- Swift diagnostics should be a single phrase or sentence, with no period at the end. If it's important to include a second idea in the diagnostic, use a semicolon to separate the two parts.
+
+- Swift diagnostics do not follow strict English grammar rules. Omit words that make the diagnostic longer without adding information. This includes grammatical words like "the".
```

Comment:
This is a sub-optimal way to phrase a very specific idea. Although it may be clear to a native speaker, an alternative reading would be that "anything goes" in terms of grammar, which I'm sure is not what you're suggesting.

There _are_, in fact, rules regarding what's going on in terms of omission of articles such as "the" in Swift diagnostics. The term you're looking for is [reduced written register](http://folk.ntnu.no/andrewww/ICL-weir-pres-handout.pdf), which is used in scenarios such as recipes ("Place noodles into pot") and news headlines.